pkg/common/cns-lib/volume/manager.go

```diff
@@ -496,6 +537,68 @@ func (m *defaultManager) createVolumeWithImprovedIdempotency(ctx context.Context
 	return resp, faultType, err
 }
 
+func (m *defaultManager) waitOnTask(ctx context.Context, taskMoRef vim25types.ManagedObjectReference) (*vim25types.TaskInfo, error) {
+	listViewCtx := context.TODO()
+	log := logger.GetLogger(listViewCtx)
+	if m.taskViewManager == nil {
+		log.Infof("wait on task manager is nil")
+		err := m.initTaskViewManager(listViewCtx)
+		if err != nil {
+			return nil, err
+		}
+	}
+	ch := m.taskViewManager.GetNewResultChannelForCaller()
+	err := m.taskViewManager.AddTaskToListView(listViewCtx, taskMoRef, ch)
+	if err != nil {
+		msg := "failed to add task to list view. err: %v"
+		log.Errorf(msg, err)
+		return nil, fmt.Errorf(msg, err)
+	}
+
+	taskInfo, err := waitForResultOrTimeout(ctx, ch)
+
+	errRemove := m.taskViewManager.RemoveTaskFromListView(listViewCtx, taskMoRef)
+	if errRemove != nil {
+		log.Errorf("failed to remove task from list view. err: %v", err)
+		err := m.taskViewManager.MarkForDeletion(listViewCtx, taskMoRef)
+		if err != nil {
+			log.Errorf("failed to mark task for deletion. err: %v", err)
+		}
+	}
+	log.Infof("pending tasks count: %+v", m.taskViewManager.GetPendingTasksCount())
+	return taskInfo, err
+}
+
+func waitForResultOrTimeout(ctx context.Context, ch chan Result) (*vim25types.TaskInfo, error) {
```

Comment:
What is the timeout value? can you explain how the timeout part works here?

Reply:
Added an explanation in the code.